### PR TITLE
Change message type to type. Type do not have fallback

### PR DESCRIPTION
**Pull request by kleewho**

### CODE REVIEW

File: fetch_request.go [hunk 1]
```diff
@@ -308,22 +317,26 @@ func (o *fetchOpts) fetchMessages(channels map[string]interface{}) map[string][]
...
 							}
 						default:
-							if o.WithMessageType {
-								histItem.MessageType = PNMessageTypeMessage.toMessageType()
+							o.pubnub.Config.Log.Printf("histResponse message_type type %vv", d)
+							if v != nil {
+								o.pubnub.Config.Log.Printf("histResponse message_type type %vv", reflect.TypeOf(v).Kind())
 							} else {
-								histItem.MessageType = ""
+								o.pubnub.Config.Log.Printf("histResponse message_type nil")
```
Comment (Xavrax) [inline fetch_request.go:333]: Is that println expected?

File: subscription_manager.go [hunk 2]
```diff
@@ -725,7 +719,7 @@ func processSubscribePayload(m *SubscriptionManager, payload subscribeMessage) {
 	}
 }
 
-func createPNFilesEvent(filePayload interface{}, m *SubscriptionManager, actualCh, subscribedCh, channel, subscriptionMatch, issuingClientID string, userMetadata interface{}, timetoken int64, messageType MessageType, spaceId SpaceId) *PNFilesEvent {
+func createPNFilesEvent(filePayload interface{}, m *SubscriptionManager, actualCh, subscribedCh, channel, subscriptionMatch, issuingClientID string, userMetadata interface{}, timetoken int64, typ string, spaceId SpaceId) *PNFilesEvent {
```
Comment (Xavrax) [inline subscription_manager.go:722]: Is that a typo?
```suggestion
func createPNFilesEvent(filePayload interface{}, m *SubscriptionManager, actualCh, subscribedCh, channel, subscriptionMatch, issuingClientID string, userMetadata interface{}, timetoken int64, type string, spaceId SpaceId) *PNFilesEvent {
```

File: subscription_manager.go [hunk 3]
```diff
@@ -757,7 +751,7 @@ func createPNFilesEvent(filePayload interface{}, m *SubscriptionManager, actualC
 		Publisher:         issuingClientID,
 		UserMetadata:      userMetadata,
 		SpaceId:           spaceId,
-		MessageType:       messageType,
+		Type:              typ,
```
Comment (Xavrax) [inline subscription_manager.go:754]: same here?
```suggestion
		Type:              type,
```

**Review by Xavrax — APPROVED** [review]
LGTM - few questions

File: subscription_manager.go (hunk 2 above)
Comment (kleewho) [reply subscription_manager.go:722]: Nope. I had plenty of options to choose from, but `type` is not one of them. It's a reserved keyword so we need to be creative with words :)

File: subscription_manager.go (hunk 3 above)
Comment (kleewho) [reply subscription_manager.go:754]: As stated in previous comment. Not typo. Keyword. That's just life :D

File: fetch_request.go (hunk 1 above)
Comment (kleewho) [reply fetch_request.go:333]: I'll check that

Comment (kleewho) [reply fetch_request.go:333]: Apparently this is how this code worked before. I'm not actually changing that. This is how it looks like straight on master

File: subscription_manager.go (hunk 3 above)
Comment (kleewho) [reply subscription_manager.go:754]: Wait. This actually look like a mistake

**Pull request merged**